src/zenml/steps/utils.py

```diff
@@ -411,6 +411,8 @@ def Do(
         if return_type is not None:
             if isinstance(return_type, Output):
                 # Resolve named (and multi-) outputs.
+                if len(list(return_type)) == 1:
```

Comment:
Something I forgot to mention in my last review: the last time I tested your branch, this change raised an exception for me, because the `Output` type is not iterable. Maybe I'm missing some other changes that landed in main in the meantime, but you might want to check on your side.

Reply:
`list(return_type.items())` instead of `list(return_type)` fixed it for me btw